flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/expressions/resolver/ExpressionResolver.java

```diff
@@ -97,15 +89,13 @@
 			ResolverRules.EXPAND_COLUMN_FUNCTIONS,
 			ResolverRules.OVER_WINDOWS,
 			ResolverRules.FIELD_RESOLVE,
-			ResolverRules.FLATTEN_CALL,
 			ResolverRules.QUALIFY_BUILT_IN_FUNCTIONS,
-			ResolverRules.RESOLVE_CALL_BY_ARGUMENTS);
+			ResolverRules.RESOLVE_CALL_BY_ARGUMENTS,
+			ResolverRules.FLATTEN_CALL);
```

Comment:
I am not sure if it is correct to call the `FLATTEN_CALL` after `RESOLVE_CALL_BY_ARGUMENTS`. The `FLATTEN_CALL` may change the arguments list of other calls.